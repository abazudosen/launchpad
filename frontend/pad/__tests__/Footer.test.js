/* @flow */

import React from 'react';
import Footer from '../Footer';
import renderer from 'react-test-renderer';

function renderFooter({ isDraft, isDeployed, canSeeLogs, url, error }) {
  return (
    <Footer
      {...{ isDraft, isDeployed, canSeeLogs, url, error }}
      isLogOpen={false}
      onResetLinkClick={() => {}}
      onLogOpen={() => {}}
      onLogClose={() => {}}
      onModalOpen={() => {}}
    />
  );
}

describe('Footer', () => {
  test('renders normal', () => {
    let component = renderer.create(
      renderFooter({
        isDraft: false,
        isDeployed: true,
        canSeeLogs: true,
        url: 'http://example.com',
        error: null,
      }),
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      renderFooter({
        isDraft: false,
        isDeployed: true,
        canSeeLogs: false,
        url: 'http://example.com',
        error: null,
      }),
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders draft', () => {
    let component = renderer.create(
      renderFooter({
        isDraft: true,
        isDeployed: true,
        canSeeLogs: true,
        url: 'http://example.com',
        error: null,
      }),
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      renderFooter({
        isDraft: true,
        isDeployed: false,
        canSeeLogs: true,
        url: 'http://example.com',
        error: null,
      }),
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders undeployed', () => {
    const component = renderer.create(
      renderFooter({
        isDraft: true,
        isDeployed: false,
        canSeeLogs: false,
        url: 'http://example.com',
        error: 'error',
      }),
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders error', () => {
    let component = renderer.create(
      renderFooter({
        isDraft: true,
        isDeployed: true,
        canSeeLogs: true,
        url: 'http://example.com',
        error: 'error',
      }),
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      renderFooter({
        isDraft: false,
        isDeployed: false,
        canSeeLogs: true,
        url: 'http://example.com',
        error: 'error',
      }),
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      renderFooter({
        isDraft: false,
        isDeployed: true,
        canSeeLogs: true,
        url: 'http://example.com',
        error: 'error',
      }),
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
