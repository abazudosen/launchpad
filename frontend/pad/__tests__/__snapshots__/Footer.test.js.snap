// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders draft 1`] = `
<div
  className="Footer"
>
  <div
    className="Footer-Left"
  >
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-lock"
      />
      <span>
        Secrets
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-paperclip"
      />
      <span>
        npm Deps
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-list-unordered"
      />
      Logs
    </div>
  </div>
  <div
    className="Footer-ErrorWrapper Footer-ErrorWrapper--yellow"
  >
    <code
      className="Footer-ErrorText"
    >
      Editing draft, save to deploy changes or
       
      <a
        href="#"
        onClick={[Function]}
      >
        reset
      </a>
       
      to saved version.
    </code>
  </div>
  <div
    className="Footer-Right"
  >
    <div
      className="footer-button"
    >
      <span
        className="icon-book"
      />
      <a
        href="https://github.com/apollographql/awesome-launchpad/"
      >
        Docs
      </a>
    </div>
    <div
      className="powered-by"
    >
      <span>
        Powered by
      </span>
      <a
        href="https://auth0.com/extend/developers"
      >
        <img
          alt="Auth0 Extend Logo"
          className="extend-logo"
          src="auth0-extend-logo.svg"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer renders draft 2`] = `
<div
  className="Footer"
>
  <div
    className="Footer-Left"
  >
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-lock"
      />
      <span>
        Secrets
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-paperclip"
      />
      <span>
        npm Deps
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-list-unordered"
      />
      Logs
    </div>
  </div>
  <div
    className="Footer-Right"
  >
    <div
      className="footer-button"
    >
      <span
        className="icon-book"
      />
      <a
        href="https://github.com/apollographql/awesome-launchpad/"
      >
        Docs
      </a>
    </div>
    <div
      className="powered-by"
    >
      <span>
        Powered by
      </span>
      <a
        href="https://auth0.com/extend/developers"
      >
        <img
          alt="Auth0 Extend Logo"
          className="extend-logo"
          src="auth0-extend-logo.svg"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer renders error 1`] = `
<div
  className="Footer"
>
  <div
    className="Footer-Left"
  >
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-lock"
      />
      <span>
        Secrets
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-paperclip"
      />
      <span>
        npm Deps
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-list-unordered"
      />
      Logs
    </div>
  </div>
  <div
    className="Footer-ErrorWrapper"
  >
    <code
      className="Footer-ErrorText"
    >
      error
       
      <span>
        Click to
         
        <a
          href="#"
          onClick={[Function]}
        >
          reset
        </a>
         
        to saved version.
      </span>
    </code>
  </div>
  <div
    className="Footer-Right"
  >
    <div
      className="footer-button"
    >
      <span
        className="icon-book"
      />
      <a
        href="https://github.com/apollographql/awesome-launchpad/"
      >
        Docs
      </a>
    </div>
    <div
      className="powered-by"
    >
      <span>
        Powered by
      </span>
      <a
        href="https://auth0.com/extend/developers"
      >
        <img
          alt="Auth0 Extend Logo"
          className="extend-logo"
          src="auth0-extend-logo.svg"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer renders error 2`] = `
<div
  className="Footer"
>
  <div
    className="Footer-Left"
  >
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-lock"
      />
      <span>
        Secrets
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-paperclip"
      />
      <span>
        npm Deps
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-list-unordered"
      />
      Logs
    </div>
  </div>
  <div
    className="Footer-ErrorWrapper"
  >
    <code
      className="Footer-ErrorText"
    >
      error
       
    </code>
  </div>
  <div
    className="Footer-Right"
  >
    <div
      className="footer-button"
    >
      <span
        className="icon-book"
      />
      <a
        href="https://github.com/apollographql/awesome-launchpad/"
      >
        Docs
      </a>
    </div>
    <div
      className="powered-by"
    >
      <span>
        Powered by
      </span>
      <a
        href="https://auth0.com/extend/developers"
      >
        <img
          alt="Auth0 Extend Logo"
          className="extend-logo"
          src="auth0-extend-logo.svg"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer renders error 3`] = `
<div
  className="Footer"
>
  <div
    className="Footer-Left"
  >
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-lock"
      />
      <span>
        Secrets
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-paperclip"
      />
      <span>
        npm Deps
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-list-unordered"
      />
      Logs
    </div>
  </div>
  <div
    className="Footer-ErrorWrapper"
  >
    <code
      className="Footer-ErrorText"
    >
      error
       
      <span>
        Click to
         
        <a
          href="#"
          onClick={[Function]}
        >
          reset
        </a>
         
        to saved version.
      </span>
    </code>
  </div>
  <div
    className="Footer-Right"
  >
    <div
      className="footer-button"
    >
      <span
        className="icon-book"
      />
      <a
        href="https://github.com/apollographql/awesome-launchpad/"
      >
        Docs
      </a>
    </div>
    <div
      className="powered-by"
    >
      <span>
        Powered by
      </span>
      <a
        href="https://auth0.com/extend/developers"
      >
        <img
          alt="Auth0 Extend Logo"
          className="extend-logo"
          src="auth0-extend-logo.svg"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer renders normal 1`] = `
<div
  className="Footer"
>
  <div
    className="Footer-Left"
  >
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-lock"
      />
      <span>
        Secrets
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-paperclip"
      />
      <span>
        npm Deps
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-list-unordered"
      />
      Logs
    </div>
  </div>
  <div
    className="Footer-EndpointURL"
  >
    <span>
      GraphQL Endpoint:
    </span>
    <span
      className="EndpointURL"
    >
      http://example.com
    </span>
  </div>
  <div
    className="Footer-Right"
  >
    <div
      className="footer-button"
    >
      <span
        className="icon-book"
      />
      <a
        href="https://github.com/apollographql/awesome-launchpad/"
      >
        Docs
      </a>
    </div>
    <div
      className="powered-by"
    >
      <span>
        Powered by
      </span>
      <a
        href="https://auth0.com/extend/developers"
      >
        <img
          alt="Auth0 Extend Logo"
          className="extend-logo"
          src="auth0-extend-logo.svg"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer renders normal 2`] = `
<div
  className="Footer"
>
  <div
    className="Footer-Left"
  >
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-lock"
      />
      <span>
        Secrets
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-paperclip"
      />
      <span>
        npm Deps
      </span>
    </div>
    <div
      className="footer-button disabled"
      onClick={[Function]}
    >
      <span
        className="icon-list-unordered"
      />
      Logs
    </div>
  </div>
  <div
    className="Footer-EndpointURL"
  >
    <span>
      GraphQL Endpoint:
    </span>
    <span
      className="EndpointURL"
    >
      http://example.com
    </span>
  </div>
  <div
    className="Footer-Right"
  >
    <div
      className="footer-button"
    >
      <span
        className="icon-book"
      />
      <a
        href="https://github.com/apollographql/awesome-launchpad/"
      >
        Docs
      </a>
    </div>
    <div
      className="powered-by"
    >
      <span>
        Powered by
      </span>
      <a
        href="https://auth0.com/extend/developers"
      >
        <img
          alt="Auth0 Extend Logo"
          className="extend-logo"
          src="auth0-extend-logo.svg"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer renders undeployed 1`] = `
<div
  className="Footer"
>
  <div
    className="Footer-Left"
  >
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-lock"
      />
      <span>
        Secrets
      </span>
    </div>
    <div
      className="footer-button"
      onClick={[Function]}
    >
      <span
        className="icon-paperclip"
      />
      <span>
        npm Deps
      </span>
    </div>
    <div
      className="footer-button disabled"
      onClick={[Function]}
    >
      <span
        className="icon-list-unordered"
      />
      Logs
    </div>
  </div>
  <div
    className="Footer-ErrorWrapper"
  >
    <code
      className="Footer-ErrorText"
    >
      error
       
    </code>
  </div>
  <div
    className="Footer-Right"
  >
    <div
      className="footer-button"
    >
      <span
        className="icon-book"
      />
      <a
        href="https://github.com/apollographql/awesome-launchpad/"
      >
        Docs
      </a>
    </div>
    <div
      className="powered-by"
    >
      <span>
        Powered by
      </span>
      <a
        href="https://auth0.com/extend/developers"
      >
        <img
          alt="Auth0 Extend Logo"
          className="extend-logo"
          src="auth0-extend-logo.svg"
        />
      </a>
    </div>
  </div>
</div>
`;
